### PR TITLE
Add feature of setting fetchmail --daemon parameter separately

### LINKED ISSUE #2567: [FR] setting fetchmail --daemon parameter separately
# Feature Request

## Context

When `FETCHMAIL_PARALLEL=1`, fetchmailrc will be split and multiple instances will be run. If we want to pass different `--daemon` parameter to fetchmail instances, we should make some change.

### Is your Feature Request related to a Problem?

No.

### Describe the Solution you'd like

With a minimal change to achive.

### Are you going to implement it?

Yes, because I know the probability of someone else doing it is low and I can learn from it.

### What are you going to contribute?

Code and docs updating.

### Alternatives you've considered

None.

### Who will that Feature be useful to?

Who has `FETCHMAIL_PARALLEL=1` in env.

### What have you done already?

Obtain the `--daemon` parameter from comment lines in fetchmailrc.

### PR DESCRIPTION
**Pull request by tinkernels**
# Description

Fixes #2567

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (README.md or the documentation under `docs/`)
- [x] If necessary I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by tinkernels** [comment]
I noticed this pull request didn't pass the test in GitHub Action.
Since I'm not familar with bats, please let me know what's the problem.

**Review by polarathene — CHANGES_REQUESTED** [review]
This is a :-1: from me.

- Your issue (feature request) and PR have not clearly justified the need for the feature.
- The current approach to implement it is not suitable for maintainers and may introduce a regression by reverting the comment stripping.

I'm also a little reluctant towards adding niche features, especially when the value is unclear. I would prefer that we have more users wanting the feature or enabling you to more easily maintain the changes on your end, similar to `user-patches.sh`.

I am open towards a different approach that doesn't use a comment to configure it. Otherwise, I think we could extract out `_setup_fetchmail_parallel` to a separate file that is sourced, and you could override that, since presently you're the only user that seems to want this feature?

File: test/config/fetchmail.cf
Comment (tinkernels) [reply test/config/fetchmail.cf:13]: > This is a 👎 from me.
> 
>     * Your issue (feature request) and PR have not clearly justified the need for the feature.
> 
>     * The current approach to implement it is not suitable for maintainers and may introduce a regression by reverting the comment stripping.
> 
> 
> I'm also a little reluctant towards adding niche features, especially when the value is unclear. I would prefer that we have more users wanting the feature or enabling you to more easily maintain the changes on your end, similar to `user-patches.sh`.
> 
> I am open towards a different approach that doesn't use a comment to configure it. Otherwise, I think we could extract out `_setup_fetchmail_parallel` to a separate file that is sourced, and you could override that, since presently you're the only user that seems to want this feature?

First, 👎 emoji looks too friendly, thanks.
As I'm trying to be a new contributor, and not familar with what you've done, thanks for pointing out problems in this pull request.
If there is a disaggressive way to achieve my goal, please let me know, maybe I can implement it the right way.

**Comment by tinkernels** [comment]
> This is a 👎 from me.
> 
>     * Your issue (feature request) and PR have not clearly justified the need for the feature.
> 
>     * The current approach to implement it is not suitable for maintainers and may introduce a regression by reverting the comment stripping.
> 
> 
> I'm also a little reluctant towards adding niche features, especially when the value is unclear. I would prefer that we have more users wanting the feature or enabling you to more easily maintain the changes on your end, similar to `user-patches.sh`.
> 
> I am open towards a different approach that doesn't use a comment to configure it. Otherwise, I think we could extract out `_setup_fetchmail_parallel` to a separate file that is sourced, and you could override that, since presently you're the only user that seems to want this feature?

First, 👎 emoji looks too friendly, thanks.
As I'm trying to be a new contributor, and not familar with what you've done, thanks for pointing out problems in this pull request.
If there is a disaggressive way to achieve my goal, please let me know, maybe I can implement it the right way.

**Comment by polarathene** [comment]
> First, :-1: emoji looks too friendly, thanks.

Assuming that's sarcasm, that was to summarise my review. I expanded after that as to why, for both you and other reviewers.

You had not received a review for almost a week, and I thought I would take the time to provide one for you :)

---

> As I'm trying to be a new contributor, and not familar with what you've done, thanks for pointing out problems in this pull request.
> If there is a disaggressive way to achieve my goal, please let me know

Apologies if I came off harsh/dismissive, that's unintentional :sweat_smile:

I was focused on the technical review only. This project is quite a lot of work for the maintainers volunteering their time regularly to keep it working well. It is common to receive feedback and iterate on contributions before they can be merged. You're not expected to know everything, but I do need to make note of changes required during review :sweat_smile: 

I think it's great that you're making a contribution to the project but at the same time I think the concerns I raised was fair?:

- Your feature request issue and this PR did not clearly describe why this feature would be helpful to users beyond your own needs.
- Your approach with parameters in the config file is non-standard, and not friendly to unload onto maintainers to look after going forward.
- It would also be helpful if more users were interested in such a feature. With your feature request issue, other users may express interest over time.
- You made a change to support your feature that may break a previous fix.

---

> maybe I can implement it the right way.

Sure :+1: 

1. Please describe a use-case for this feature and why it would be useful for other users.
2. Wait for other maintainers to chime in. The feature should wait for approval to continue, otherwise your time would be wasted.
3. If approved, we can discuss a better approach for you to try. Otherwise, we can discuss the possibility to make it easier for you to maintain your own patch.

You may not get the outcome that you'd like, and I can relate to that when I contribute to other projects myself. I hope you can understand the concerns that have been expressed are specific to the earlier concerns raised. Contributions are very welcome, and your time appreciated, but unfortunately sometimes not suitable to accept.

---

So for now, please wait for another review/feedback, but it would be great if you could describe why this feature is important.

**Pull request closed without merging**
